core/trino-main/src/main/java/io/trino/sql/planner/DomainTranslator.java

```diff
@@ -440,6 +440,13 @@ protected ExtractionResult visitNotExpression(NotExpression node, Boolean comple
             return process(node.getValue(), !complement);
         }
 
+        @Override
+        protected ExtractionResult visitSymbolReference(SymbolReference node, Boolean complement)
+        {
+            ComparisonExpression newNode = new ComparisonExpression(EQUAL, node, TRUE_LITERAL);
```

Comment:
you should only apply this for `boolean` symbols. It doesn't make sense for non boolean symbols